CHANGELOG.rst

```diff
@@ -1,3 +1,41 @@
+pytest-xdist 3.6.0 (2024-04-19)
+===============================
+
+Bug Fixes
+---------
+
+- `#1024 <https://github.com/pytest-dev/pytest-xdist/issues/1024>`_: Added proper handling of ``shouldstop`` (such as set by ``--max-fail``) and ``shouldfail`` conditions in workers.
+  Previously, a worker might have continued executing further tests before the controller could terminate the session.
+
+- `#1028 <https://github.com/pytest-dev/pytest-xdist/issues/1028>`_: Fix compatiblity issue between `looponfail` and editable installs.
+
+- `#620 <https://github.com/pytest-dev/pytest-xdist/issues/620>`_: Use the ``execnet`` new ``main_thread_only`` "execmodel" so that code which expects to only run in the main thread will now work as expected.
+
+- `#937 <https://github.com/pytest-dev/pytest-xdist/issues/937>`_: Fix a bug where plugin would raise an incompatibility error with ``--pdb`` despite using ``-n0``.
```

Comment:
Minor adjustments.

```suggestion
- `#1024 <https://github.com/pytest-dev/pytest-xdist/issues/1024>`_: Added proper handling of ``shouldstop`` (such as set by ``--max-fail``) and ``shouldfail`` conditions in workers.
  Previously, a worker might have continued executing further tests before the controller could terminate the session.

- `#1028 <https://github.com/pytest-dev/pytest-xdist/issues/1028>`_: Fixed compatiblity issue between `looponfail` and editable installs.

- `#620 <https://github.com/pytest-dev/pytest-xdist/issues/620>`_: Use the new  ``main_thread_only`` ``execnet`` "execmodel" so that code which expects to only run in the main thread will now work as expected.

- `#937 <https://github.com/pytest-dev/pytest-xdist/issues/937>`_: Fixed a bug where plugin would raise an incompatibility error with ``--pdb`` despite using ``-n0``.
```